### PR TITLE
Falling Falcons: Paratroopers Edition

### PR DESCRIPTION
**Pull request by ihatethisengine**
# About the pull request
Replaces rappel system with paradrop system. You need a parachute on your back to paradrop (successfully). You also can drop objects from the above, but in most cases it's not a great idea (they can break). Crates can be paradropped too, so this can be an interesting alternative to req drops. In order for marines to paradrop, PO needs to choose a flare on the console and lock the paradrop system to the signal. After that the aft hatch will be open and you are free to jump! Just don't forget the parachute. (SGs can wear parachutes with armor)

Credits to @SubjectD9341 for the sprites.

# Explain why it's good for the game
It's cooler than rappel.

# Testing Photographs and Procedure
<details>

https://github.com/cmss13-devs/cmss13/assets/115417687/53ec9ddb-5ff5-4af2-9df5-17835e5bc2bc

https://github.com/cmss13-devs/cmss13/assets/115417687/f296c1c7-1cfe-4f8f-89c0-233f612d8542

https://github.com/cmss13-devs/cmss13/assets/115417687/579dac1c-9248-4a7c-b0f7-c9466a131335

https://github.com/cmss13-devs/cmss13/assets/115417687/35728d57-7c83-4539-bf84-accb99127ad7

</details>

# Changelog
:cl: ihatethisengine
add: Added paradropping
del: Removed rappeling
/:cl:

### CODE REVIEW

**Review by fira — DISMISSED** [review]

File: code/game/turfs/transit.dm [hunk 1]
```diff
@@ -66,54 +98,109 @@
...
-		// I'd say falling from space is pretty much like getting hit by an explosion
-		human.take_overall_armored_damage(300, ARMOR_BOMB, limb_damage_chance = 100)
-		// but just in case, you will still take a ton of damage.
-		human.take_overall_damage(200, used_weapon = "falling", limb_damage_chance = 100)
-		if(human.stat != DEAD)
-			human.death(human.last_damage_data)
-		fallen_mob.status_flags |= PERMANENTLY_DEAD
-		return
+	KnockDown(10) // 10 seconds
+	Stun(3) // 3 seconds
```
[the remark this answers is not in the document]
Comment (ihatethisengine) [reply code/game/turfs/transit.dm:187]: it doesnt really matter you are dead

File: tgui/packages/tgui/interfaces/MfdPanels/ParadropPanel.tsx [hunk 2]
```diff
@@ -0,0 +1,76 @@
...
+              {paradropData.signal
+                ? 'Locked to ' + paradropData.signal + '.'
+                : 'No locked target found.'}
+            </h3>
+          </Stack.Item>
+          <Stack.Item>
+            <h3>
+              {paradropData.locked
+                ? 'Paradropping available.'
+                : 'Paradropping not available.'}
```
[the remark this answers is not in the document]
Comment (ihatethisengine) [reply tgui/packages/tgui/interfaces/MfdPanels/ParadropPanel.tsx:31]: why did you resolve it it's a fair question

**Comment by ihatethisengine** [comment]
My parachute sprites are being held hostage but I should sort it out soon

**Comment by LC4492** [comment]
As I commented before down on discord in the contributor's channel, wouldn't be better if the marines also had the possibility of attaching the parachute on their armor (like a poncho), so they wouldn't need to sacrifice their back slot? Good PR anyway

**Comment by ihatethisengine** [comment]
> As I commented before down on discord in the contributor's channel, wouldn't be better if the marines also had the possibility of attaching the parachute on their armor (like a poncho), so they wouldn't need to sacrifice their back slot? Good PR anyway

You don't really need to sacrifice anything, you can jump with items in your hands, just like you could with rappel.

File: code/__DEFINES/equipment.dm [hunk 3]
```diff
@@ -84,6 +84,8 @@
 #define IGNITING_ITEM (1<<13)
 /// Overrides NODROP in some cases (stripping)
 #define FORCEDROP_CONDITIONAL (1<<14)
+/// Overrides smartgunner no being able to wear backpacks
```
Comment (Drulikar) [inline code/__DEFINES/equipment.dm:87]:
```suggestion
/// Overrides smartgunner not being able to wear backpacks
```

File: code/game/turfs/transit.dm [hunk 4]
```diff
@@ -39,14 +41,42 @@
...
+			var/found_dense = FALSE
+			for(var/atom/A in T)
+				if(A.density && A.can_block_movement)
+					found_dense = TRUE
+					break
+			if(found_dense)
+				continue
+			if(protected_by_pylon(TURF_PROTECTION_MORTAR, T))
+				continue
+			valid_turfs += T
```
Comment (Drulikar) [inline code/game/turfs/transit.dm:67]:
```suggestion
		for(var/turf/turf as anything in RANGE_TURFS(crosser.get_paradrop_scatter(), location))
			var/area/turf_area = get_area(turf)
			if(!turf_area || CEILING_IS_PROTECTED(turf_area.ceiling, CEILING_PROTECTION_TIER_1))
				continue
			if(turf.density)
				continue
			var/found_dense = FALSE
			for(var/atom/turf_atom in turf)
				if(turf_atom.density && turf_atom.can_block_movement)
					found_dense = TRUE
					break
			if(found_dense)
				continue
			if(protected_by_pylon(TURF_PROTECTION_MORTAR, turf))
				continue
			valid_turfs += turf
```

File: code/game/turfs/transit.dm [hunk 5]
```diff
@@ -67,53 +97,121 @@
...
+	return TRUE
+
+/obj/structure/closet/crate/get_paradrop_scatter() //crates land closer to the signal
+	return 4
+
+/atom/movable/proc/handle_paradrop(turf/target, dropship_name)
+	ADD_TRAIT(src, TRAIT_IMMOBILIZED, TRAIT_SOURCE_DROPSHIP_INTERACTION)
+	ADD_TRAIT(src, TRAIT_UNDENSE, TRAIT_SOURCE_DROPSHIP_INTERACTION)
+	var/image/cables = image('icons/obj/structures/droppod_32x64.dmi', src, "chute_cables_static")
+	src.overlays += cables
```
Comment (Drulikar) [inline code/game/turfs/transit.dm:135]:
```suggestion
	overlays += cables
```

File: code/game/turfs/transit.dm [hunk 6]
```diff
@@ -67,53 +97,121 @@
...
+	var/image/chute = image('icons/obj/structures/droppod_64x64.dmi', src, "chute_static")
+
+	chute.pixel_x -= 16
+	chute.pixel_y += 16
+
+	src.overlays += chute
+	src.pixel_z = 360
+	src.forceMove(target)
```
Comment (Drulikar) [inline code/game/turfs/transit.dm:143]:
```suggestion
	overlays += chute
	pixel_z = 360
	forceMove(target)
```

File: code/game/turfs/transit.dm [hunk 7]
```diff
@@ -67,53 +97,121 @@
...
-		return // don't know how you got here, but you shouldnt be here.
-	var/mob/living/fallen_mob = crosser
+/obj/item/handle_airdrop(turf/target, dropship_name)
+	..()
+	if(QDELETED(src))
+		return
+	if(!indestructible && w_class == SIZE_SMALL) //small items will be lost, good riddance
+		deconstruct(FALSE)
+		return
+	src.explosion_throw(200) // give it a bit of a kick
```
Comment (Drulikar) [inline code/game/turfs/transit.dm:189]:
```suggestion
	explosion_throw(200) // give it a bit of a kick
```

**Review by Drulikar — CHANGES_REQUESTED** [review]

File: code/game/turfs/transit.dm [hunk 8]
```diff
@@ -67,53 +97,121 @@
...
 
-	REMOVE_TRAIT(crosser, TRAIT_IMMOBILIZED, TRAIT_SOURCE_DROPSHIP_INTERACTION)
-	if(isitem(crosser))
-		var/obj/item/item = crosser
-		item.explosion_throw(200) // give it a bit of a kick
+/obj/structure/closet/crate/can_paradrop() //for now all crates can be paradropped
+	return TRUE
+
+/obj/structure/closet/crate/get_paradrop_scatter() //crates land closer to the signal
+	return 4
```
Comment (Drulikar) [inline code/game/turfs/transit.dm:129]: Please also do the same for `/obj/structure/largecrate`

**Review by Drulikar — CHANGES_REQUESTED** [review]
Also consider expending the parachute on land; or put it in a state where you have to re-pack it.

File: code/game/turfs/transit.dm [hunk 9]
```diff
@@ -67,53 +97,127 @@
...
+	if(!opened)
+		for(var/atom/movable/content in src)
+			INVOKE_ASYNC(content, TYPE_PROC_REF(/atom/movable, handle_airdrop), target, dropship_name)
+		open()
+	. = ..()
 
-	if(!isliving(crosser))
-		return // don't know how you got here, but you shouldnt be here.
-	var/mob/living/fallen_mob = crosser
+/obj/item/handle_airdrop(turf/target, dropship_name)
```
Comment (Drulikar) [inline code/game/turfs/transit.dm:188]: Also please have primed grenades or planted c4 just qdel themselves as if they exploded mid-air including custom grenades even if they may have set a long timer.

**Review by Drulikar — CHANGES_REQUESTED** [review]

File: code/game/turfs/transit.dm (hunk 9 above)
Comment (ihatethisengine) [reply code/game/turfs/transit.dm:196]: done, tested by jumping with active hefa

**Review by Drulikar — APPROVED** [review]

**Review by nauticall — APPROVED** [review]

**Pull request merged**